// Copyright ICR
//

syntax = "proto3";
package pyosirix;

message ROIListRequest {
  string id = 1;
}

message Point {
  float x = 1;
  float y = 2;
  float z = 3;
}

message Point2D {
  float x = 1;
  float y = 2;
}

message Color {
  uint32 r = 1;
  uint32 g = 2;
  uint32 b = 3;
}

message ROI {
  string id = 1;  // patient id :: seriesinstanceUID :: slice number
  string name = 2;
  string series_uid = 3;
  string image_uid = 4;
  Color color = 5;
  float opacity = 6;
  float thickness = 7;
  uint32 selected = 8;
  Point centre = 9;
  Point2D offset = 10;
  Point2D offset_between_mm = 11; // set z levels in mm for which to apply the offset (inclusive)
  repeated Point points = 12;
}

// message SelectedROIResponse {
//   string id = 1;  // patient id :: seriesinstanceUID :: slice number
//   repeated ROI roi_list = 2;
// }

message ROIListResponse {
  string id = 1;  // patient id :: seriesinstanceUID :: slice number
  repeated ROI roi_list = 2;
}

message ROIImageRequest {
  string id = 1;
  string input_filepath = 2;
  //int32 fill_value = 2;
}

message ROIImageResponse {
  string id = 1;
  string output_filepath = 2;
}
